pkg/broker/handler/pool.go

```diff
@@ -80,6 +86,13 @@ func (c *probeChecker) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		w.WriteHeader(http.StatusNotFound)
 		return
 	}
+	// Perform Authentication check.
+	if err := authcheck.AuthenticationCheck(req.Context(), c.authType, c.authCheckClient); err != nil {
+		c.logger.Error("authentication check failed", zap.Error(err))
+		w.WriteHeader(http.StatusUnauthorized)
```

Comment:
Is it a common practice to return 403 in prober? I find it's confusing because it's hard to tell whether it's "called not allowed to do the health check" or "health check failed".

Reply:
I think the probe result is showing the pod status, for whether it is having issues while the process is running. So I choose 401 to indicate the process is running with auth problem. I think this is the most close one I could find, but I could be wrong. Which statusCode do you think is more proper?

Reply:
4xx means client error so its natural meaning is that kubelet is not authorized to call the health check. I think 500 should be sufficient. Are you depending on the status code to do something?

Reply:
makes sense, changed